src/Provider/Mastodon.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Hybridauth\Provider;
+
+use Hybridauth\Adapter\OAuth2;
+use Hybridauth\Exception\InvalidApplicationCredentialsException;
+use Hybridauth\Exception\UnexpectedApiResponseException;
+use Hybridauth\Data;
+use Hybridauth\User\Profile;
+
+class Mastodon extends OAuth2
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public $scope = 'read';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $apiDocumentation = 'https://docs.joinmastodon.org/spec/oauth/';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        if (!$this->config->exists('url')) {
+            throw new InvalidApplicationCredentialsException(
+                'You must define a Mastodon instance url'
+            );
+        }
+        $url = $this->config->get('url');
+
+        $this->apiBaseUrl = $url . '/api/v1';
+
+        $this->authorizeUrl = $url . '/oauth/authorize';
+        $this->accessTokenUrl = $url . '/oauth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUserProfile()
+    {
+		$response = $this->apiRequest('accounts/verify_credentials', 'GET', []);
```

Comment:
Hi, thanks for rising PR,
in general, it looks good

can we fix some alignment issue in a few places? it should be 4 spaces